server/models/pattern/core/pattern.go

```diff
@@ -325,46 +325,84 @@ func processCytoElementsWithPattern(eles []cytoscapejs.Element, callback func(sv
-func NewPatternFileFromK8sManifest(data string, fileName string, ignoreErrors bool, reg *registry.RegistryManager) (pattern.PatternFile, error) {
-	if fileName == "" {
-		fileName = "Autogenerated"
-	}
-
-	pattern := pattern.PatternFile{
-		SchemaVersion: v1beta1.DesignSchemaVersion,
-		Name:          fileName,
-		Components:    []*component.ComponentDefinition{},
-	}
+// getAllK8sResources extracts all Kubernetes resources from the provided YAML data.
+// It handles multi-document YAML files, processes "List" kinds to extract individual items,
+// and returns a slice of resource manifests as generic map[string]interface{} structures.
+func getAllK8sResources(data string, ignoreErrors bool) ([]map[string]interface{}, error) {
+	var K8sResources []map[string]interface{}
+	decoder := yaml.NewDecoder(bytes.NewBufferString(data))
 
-	buffer := bytes.NewBufferString(data)
-	decoder := yaml.NewDecoder(buffer)
 	for {
 		manifest := map[string]interface{}{}
-
 		err := decoder.Decode(manifest)
 		if err != nil {
 			if err == io.EOF {
-				if len(pattern.Components) == 0 {
-					return pattern, ErrParseK8sManifest(fmt.Errorf("kubernetes manifest is empty"))
-				}
-				return pattern, nil
-			} else {
-				return pattern, ErrParseK8sManifest(fmt.Errorf("kubernetes manifest is empty"))
+				break
 			}
+			return K8sResources, ErrParseK8sManifest(fmt.Errorf("kubernetes manifest is invalid: %v", err))
 		}
+
+		// Skip empty manifests
 		if len(manifest) == 0 {
 			continue
 		}
-		// Recursive casting
+
 		manifest = utils.RecursiveCastMapStringInterfaceToMapStringInterface(manifest)
 		if manifest == nil {
 			if ignoreErrors {
 				continue
 			}
-			return pattern, ErrParseK8sManifest(fmt.Errorf("failed to parse manifest into an internal representation"))
+			return K8sResources, ErrParseK8sManifest(fmt.Errorf("failed to parse manifest into an internal representation"))
 		}
 
+		// If it's a List kind, expand it, otherwise add directly
+		kind, _ := mutils.Cast[string](manifest["kind"])
+		if kind == "List" {
+			if items, ok := manifest["items"].([]interface{}); ok {
+				for _, item := range items {
+					if itemMap, ok := item.(map[string]interface{}); ok {
+						K8sResources = append(K8sResources, itemMap)
+					}
+				}
+			}
+			continue
+		}
+
+		K8sResources = append(K8sResources, manifest)
+	}
+
+	return K8sResources, nil
+}
+
+// NewPatternFileFromK8sManifest converts Kubernetes manifests into a Meshery design pattern file.
+// It extracts all resources, processes each resource to generate component
+// definitions, and combines them into a design pattern. If `ignoreErrors` is true, it skips errors
+// during the creation of individual component definitions.
+//
+// Note: If modified, make sure this function always returns a meshkit error
+func NewPatternFileFromK8sManifest(data string, fileName string, ignoreErrors bool, reg *registry.RegistryManager) (pattern.PatternFile, error) {
+	if fileName == "" {
+		fileName = "Autogenerated"
```

Comment:
There should always be a filename available via URI, whether the file was presented via filesystem path, file upload, URL (http or oci). We've been meaning to get away from the "Autogenerated" name. Is there any case in which we won't have a filename available?

Reply:
I tested in the UI, seems like the UI allows to import designs without design name be it URL or by upload and since file name is not being sent it falls back to design name and since it's empty as well it fallbacks back to Autogenerated.